### ISSUE TITLE
NoSuchMethodError on POST/PATCH in 2.2.0

### ISSUE THREAD

**Issue by hamiller** [katharsis-project/katharsis-rs#31]
When trying to POST/PATCH (via curl) against a repository, I'll get an error (sse below).
Switching to 2.1.3 solves this problem.

```
ERROR [io.undertow.request] (default task-45) UT005023: Exception handling request to /ServiceRegistry/api/endpoints: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectMapper.readerFor(Ljava/lang/Class;)Lcom/fasterxml/jackson/databind/ObjectReader;
	at io.katharsis.dispatcher.controller.resource.ResourceUpsert.setAttributes(ResourceUpsert.java:55)
	at io.katharsis.dispatcher.controller.resource.ResourcePost.handle(ResourcePost.java:74)
	at io.katharsis.dispatcher.controller.resource.ResourcePost.handle(ResourcePost.java:27)
	at io.katharsis.dispatcher.RequestDispatcher.dispatchRequest(RequestDispatcher.java:46)
	at io.katharsis.rs.KatharsisFilter.dispatchRequest(KatharsisFilter.java:125)
	at io.katharsis.rs.KatharsisFilter.filter(KatharsisFilter.java:99)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:127)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:167)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:86)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:72)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:282)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:261)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:80)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:172)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:199)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:774)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

(Feature in 2.2.0 - not working):
```
@Provider
public class KatharsisFeature implements Feature {
    @Override
    public boolean configure(FeatureContext featureContext) {
        featureContext
        		.property(KatharsisProperties.RESOURCE_SEARCH_PACKAGE, "xxx")
                .register(new io.katharsis.rs.KatharsisFeature(new ObjectMapper(), new QueryParamsBuilder(new DefaultQueryParamsParser()), new WildflyServiceLocator()));
        return true;
    }
}
```

(Feature in 2.1.3 - working):
```
@Provider
public class KatharsisFeature implements Feature {
    @Override
    public boolean configure(FeatureContext featureContext) {
        featureContext
        		.property(KatharsisProperties.RESOURCE_SEARCH_PACKAGE, "xxx")
                .register(new io.katharsis.rs.KatharsisFeature(new ObjectMapper(), new WildflyServiceLocator()));
        return true;
    }
}
```

curl / httpie command 
 `echo '{ "data": { "type": "endpoints", "id": "DELETE", "attributes": { "description": "Dies ist eine laengere Erklaerung", "uri": "http://localhost/nix" } } }' | http POST http://localhost:8080/ServiceRegistry/api/endpoints Accept:application/vnd.api+json Content-Type:application/vnd.api+json`

**Comment by meshuga** [comment]
Which version of Jackson do you use?
Note that Katharsis require version 2.6.3+ to run.

**Comment by hamiller** [comment]
None: Its using the bundled Resteasy version from wildfy 9.
Also: Using UndertowJaxrsServer (1.3.17.Final) in my JUnit tests produces no error with 2.2.0 and 2.1.3

**Comment by HardyLoppmann** [comment]
I´ve got a similar problem, using Oracle WLS 12c, Java 1.8.0_60, Katharsis 2.3.0 and Jackson Mapper 2.7.1
A GET command like this 
`http :7001/recommender-0.0.1-SNAPSHOT/api/recommendation`
I got a 500er
```
####<Feb 22, 2016 8:38:00 AM CET> <Error> <HTTP> <ak-MBP.local> <AdminServer> <[ACTIVE] ExecuteThread: '7' for queue: 'weblogic.kernel.Default (self-tuning)'> <<WLS Kernel>> <> <227c524d-2d75-4dd2-b36e-0a27b79cc93e-00000037> <1456126680529> <[severity-value: 8] [rid: 0] [partition-id: 0] [partition-name: DOMAIN] > <BEA-101017> <[ServletContext@1977178398[app:recommender module:recommender-0.0.1-SNAPSHOT.war path:null spec-version:3.1], request: weblogic.servlet.internal.ServletRequestImpl@3f8f7d40[
GET /recommender-0.0.1-SNAPSHOT/api/recommendation HTTP/1.1
Connection: keep-alive
Accept-Encoding: gzip, deflate
Accept: */*
User-Agent: HTTPie/0.9.3

]] Root cause of ServletException.
org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectMapper.setFilterProvider(Lcom/fasterxml/jackson/databind/ser/FilterProvider;)Lcom/fasterxml/jackson/databind/ObjectMapper;
	at org.glassfish.jersey.servlet.internal.ResponseWriter.rethrow(ResponseWriter.java:256)
	at org.glassfish.jersey.servlet.internal.ResponseWriter.failure(ResponseWriter.java:238)
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:486)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:317)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:292)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:460)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:386)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:334)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221)
	at weblogic.servlet.internal.StubSecurityHelper$ServletServiceAction.run(StubSecurityHelper.java:286)
	at weblogic.servlet.internal.StubSecurityHelper$ServletServiceAction.run(StubSecurityHelper.java:260)
	at weblogic.servlet.internal.StubSecurityHelper.invokeServlet(StubSecurityHelper.java:137)
	at weblogic.servlet.internal.ServletStubImpl.execute(ServletStubImpl.java:350)
	at weblogic.servlet.internal.TailFilter.doFilter(TailFilter.java:25)
	at weblogic.servlet.internal.FilterChainImpl.doFilter(FilterChainImpl.java:78)
	at weblogic.servlet.internal.RequestEventsFilter.doFilter(RequestEventsFilter.java:32)
	at weblogic.servlet.internal.FilterChainImpl.doFilter(FilterChainImpl.java:78)
	at weblogic.servlet.internal.WebAppServletContext$ServletInvocationAction.wrapRun(WebAppServletContext.java:3654)
	at weblogic.servlet.internal.WebAppServletContext$ServletInvocationAction.run(WebAppServletContext.java:3620)
	at weblogic.security.acl.internal.AuthenticatedSubject.doAs(AuthenticatedSubject.java:326)
	at weblogic.security.service.SecurityManager.runAsForUserCode(SecurityManager.java:196)
	at weblogic.servlet.provider.WlsSecurityProvider.runAsForUserCode(WlsSecurityProvider.java:203)
	at weblogic.servlet.provider.WlsSubjectHandle.run(WlsSubjectHandle.java:71)
	at weblogic.servlet.internal.WebAppServletContext.doSecuredExecute(WebAppServletContext.java:2423)
	at weblogic.servlet.internal.WebAppServletContext.securedExecute(WebAppServletContext.java:2280)
	at weblogic.servlet.internal.WebAppServletContext.execute(WebAppServletContext.java:2258)
	at weblogic.servlet.internal.ServletRequestImpl.runInternal(ServletRequestImpl.java:1626)
	at weblogic.servlet.internal.ServletRequestImpl.run(ServletRequestImpl.java:1586)
	at weblogic.servlet.provider.ContainerSupportProviderImpl$WlsRequestExecutor.run(ContainerSupportProviderImpl.java:270)
	at weblogic.invocation.ComponentInvocationContextManager._runAs(ComponentInvocationContextManager.java:348)
	at weblogic.invocation.ComponentInvocationContextManager.runAs(ComponentInvocationContextManager.java:333)
	at weblogic.work.LivePartitionUtility.doRunWorkUnderContext(LivePartitionUtility.java:54)
	at weblogic.work.PartitionUtility.runWorkUnderContext(PartitionUtility.java:41)
	at weblogic.work.SelfTuningWorkManagerImpl.runWorkUnderContext(SelfTuningWorkManagerImpl.java:617)
	at weblogic.work.ExecuteThread.execute(ExecuteThread.java:397)
	at weblogic.work.ExecuteThread.run(ExecuteThread.java:346)
Caused By: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectMapper.setFilterProvider(Lcom/fasterxml/jackson/databind/ser/FilterProvider;)Lcom/fasterxml/jackson/databind/ObjectMapper;
	at io.katharsis.jackson.serializer.ContainerSerializer.getObjectMapper(ContainerSerializer.java:242)
	at io.katharsis.jackson.serializer.ContainerSerializer.writeAttributes(ContainerSerializer.java:153)
	at io.katharsis.jackson.serializer.ContainerSerializer.writeData(ContainerSerializer.java:91)
	at io.katharsis.jackson.serializer.ContainerSerializer.serialize(ContainerSerializer.java:64)
	at io.katharsis.jackson.serializer.ContainerSerializer.serialize(ContainerSerializer.java:37)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:117)
	at com.fasterxml.jackson.databind.ser.std.CollectionSerializer.serializeContents(CollectionSerializer.java:23)
	at com.fasterxml.jackson.databind.ser.std.AsArraySerializerBase.serialize(AsArraySerializerBase.java:183)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:114)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:1837)
	at com.fasterxml.jackson.core.base.GeneratorBase.writeObject(GeneratorBase.java:261)
	at com.fasterxml.jackson.core.JsonGenerator.writeObjectField(JsonGenerator.java:1312)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serializeResourceCollection(BaseResponseSerializer.java:112)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.writeResponseWithResources(BaseResponseSerializer.java:73)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serialize(BaseResponseSerializer.java:37)
	at io.katharsis.jackson.serializer.BaseResponseSerializer.serialize(BaseResponseSerializer.java:15)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:114)
	at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue(ObjectMapper.java:2818)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2241)
	at io.katharsis.rs.KatharsisFilter.abortWithResponse(KatharsisFilter.java:151)
	at io.katharsis.rs.KatharsisFilter.dispatchRequest(KatharsisFilter.java:132)
	at io.katharsis.rs.KatharsisFilter.filter(KatharsisFilter.java:99)
	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:132)
	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:68)
	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:197)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:301)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:292)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:460)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:386)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:334)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221)
	at weblogic.servlet.internal.StubSecurityHelper$ServletServiceAction.run(StubSecurityHelper.java:286)
	at weblogic.servlet.internal.StubSecurityHelper$ServletServiceAction.run(StubSecurityHelper.java:260)
	at weblogic.servlet.internal.StubSecurityHelper.invokeServlet(StubSecurityHelper.java:137)
	at weblogic.servlet.internal.ServletStubImpl.execute(ServletStubImpl.java:350)
	at weblogic.servlet.internal.TailFilter.doFilter(TailFilter.java:25)
	at weblogic.servlet.internal.FilterChainImpl.doFilter(FilterChainImpl.java:78)
	at weblogic.servlet.internal.RequestEventsFilter.doFilter(RequestEventsFilter.java:32)
	at weblogic.servlet.internal.FilterChainImpl.doFilter(FilterChainImpl.java:78)
	at weblogic.servlet.internal.WebAppServletContext$ServletInvocationAction.wrapRun(WebAppServletContext.java:3654)
	at weblogic.servlet.internal.WebAppServletContext$ServletInvocationAction.run(WebAppServletContext.java:3620)
	at weblogic.security.acl.internal.AuthenticatedSubject.doAs(AuthenticatedSubject.java:326)
	at weblogic.security.service.SecurityManager.runAsForUserCode(SecurityManager.java:196)
	at weblogic.servlet.provider.WlsSecurityProvider.runAsForUserCode(WlsSecurityProvider.java:203)
	at weblogic.servlet.provider.WlsSubjectHandle.run(WlsSubjectHandle.java:71)
	at weblogic.servlet.internal.WebAppServletContext.doSecuredExecute(WebAppServletContext.java:2423)
	at weblogic.servlet.internal.WebAppServletContext.securedExecute(WebAppServletContext.java:2280)
	at weblogic.servlet.internal.WebAppServletContext.execute(WebAppServletContext.java:2258)
	at weblogic.servlet.internal.ServletRequestImpl.runInternal(ServletRequestImpl.java:1626)
	at weblogic.servlet.internal.ServletRequestImpl.run(ServletRequestImpl.java:1586)
	at weblogic.servlet.provider.ContainerSupportProviderImpl$WlsRequestExecutor.run(ContainerSupportProviderImpl.java:270)
	at weblogic.invocation.ComponentInvocationContextManager._runAs(ComponentInvocationContextManager.java:348)
	at weblogic.invocation.ComponentInvocationContextManager.runAs(ComponentInvocationContextManager.java:333)
	at weblogic.work.LivePartitionUtility.doRunWorkUnderContext(LivePartitionUtility.java:54)
	at weblogic.work.PartitionUtility.runWorkUnderContext(PartitionUtility.java:41)
	at weblogic.work.SelfTuningWorkManagerImpl.runWorkUnderContext(SelfTuningWorkManagerImpl.java:617)
	at weblogic.work.ExecuteThread.execute(ExecuteThread.java:397)
	at weblogic.work.ExecuteThread.run(ExecuteThread.java:346)
```
the dependencys are very light for maven 
```
➜  hans git:(master) ✗ mvn dependency:tree
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Recommender 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ recommender ---
[INFO] com.hanswurst.tool:recommender:war:0.0.1-SNAPSHOT
[INFO] +- io.katharsis:katharsis-rs:jar:2.3.0:compile
[INFO] |  +- io.katharsis:katharsis-core:jar:2.3.0:compile
[INFO] |  |  \- net.jodah:typetools:jar:0.4.4:compile
[INFO] |  +- javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile
[INFO] |  \- javax.servlet:javax.servlet-api:jar:3.0.1:compile
[INFO] +- org.reflections:reflections:jar:0.9.10:compile
[INFO] |  +- com.google.guava:guava:jar:15.0:compile
[INFO] |  +- org.javassist:javassist:jar:3.19.0-GA:compile
[INFO] |  \- com.google.code.findbugs:annotations:jar:2.0.1:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.7.1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.7.0:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-core:jar:2.7.1:compile
[INFO] +- net.sf.dozer:dozer:jar:5.5.1:compile
[INFO] |  +- commons-beanutils:commons-beanutils:jar:1.9.1:compile
[INFO] |  |  \- commons-collections:commons-collections:jar:3.2.1:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.2.1:compile
[INFO] |  \- org.slf4j:jcl-over-slf4j:jar:1.7.5:compile
[INFO] +- org.codehaus.groovy:groovy-all:jar:2.4.4:compile
[INFO] +- javax:javaee-api:jar:6.0:provided
[INFO] +- javax.enterprise:cdi-api:jar:1.2:provided
[INFO] |  +- javax.el:javax.el-api:jar:3.0.0:provided
[INFO] |  +- javax.interceptor:javax.interceptor-api:jar:1.2:provided
[INFO] |  \- javax.inject:javax.inject:jar:1:provided
[INFO] +- junit:junit:jar:4.12:compile
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.13:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.1.3:compile
[INFO] |  \- ch.qos.logback:logback-core:jar:1.1.3:compile
[INFO] \- org.slf4j:jul-to-slf4j:jar:1.7.13:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.302 s
[INFO] Finished at: 2016-02-22T08:42:08+01:00
[INFO] Final Memory: 12M/245M
[INFO] ------------------------------------------------------------------------
➜  hans git:(master) ✗
```

**Comment by meshuga** [comment]
Are you sure you use Jackson version **2.6+**?

Take a look at the JavaDoc the method your apps cannot find. All of them are available since Jackson 2.6.0 and at this point of time I don't want to downgrade the Jackson library.
```java
    /**
     * Factory method for constructing {@link ObjectReader} that will
     * read or update instances of specified type
     * 
     * @since 2.6
     */
    public ObjectReader readerFor(Class<?> type) {
        ...
    }
```

```java
    /**
     * Method for configuring this mapper to use specified {@link FilterProvider} for
     * mapping Filter Ids to actual filter instances.
     *<p>
     * Note that usually it is better to use method {@link #writer(FilterProvider)};
     * however, sometimes
     * this method is more convenient. For example, some frameworks only allow configuring
     * of ObjectMapper instances and not {@link ObjectWriter}s.
     * 
     * @since 2.6
     */
    public ObjectMapper setFilterProvider(FilterProvider filterProvider) {
        ...
    }
```

**Comment by HardyLoppmann** [comment]
since an hour I´m sure that the WLS hasn´t used the Mapper 2.7.1. now the weblogic-application.xml is successfully changed
